flash/core/finetuning.py

```diff
@@ -11,147 +11,169 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-from typing import List, Union
+from functools import partial
+from typing import Iterable, Optional, Tuple, Union
 
-from pytorch_lightning import LightningModule
 from pytorch_lightning.callbacks import BaseFinetuning
-from pytorch_lightning.utilities.exceptions import MisconfigurationException
-from torch import nn
+from pytorch_lightning.utilities.enums import LightningEnum
+from torch.nn import Module
 from torch.optim import Optimizer
 
+from flash.core.registry import FlashRegistry
 
-class NoFreeze(BaseFinetuning):
-    def freeze_before_training(self, pl_module: LightningModule) -> None:
-        pass
 
-    def finetune_function(
-        self,
-        pl_module: LightningModule,
-        epoch: int,
-        optimizer: Optimizer,
-        opt_idx: int,
-    ) -> None:
-        pass
+class FinetuningStrategies(LightningEnum):
```

Comment:
I like the Enum :smiley: I think it would be neat to also keep the separate class for each strategy even if they just override init with a hard coded strategy key, that way it can be backwards compatible.

Reply:
So, should I revert the changes and map each Enum member to a class implementation ?

Reply:
No need to revert, could just have e.g.:
```
class FreezeUnfreeze(FlashBaseFinetuning):
    def __init__(
        self,
        strategy_metadata: Optional[Union[int, Tuple[int, int]]] = None,
        train_bn: bool = True,
    ):
        super().__init__(FinetuningStrategies.FREEZE_UNFREEZE, strategy_metadata, train_bn)
```

Reply:
Got it.